runtime/src/main/java/org/corfudb/runtime/view/QueueAddressManager.java

```diff
@@ -0,0 +1,345 @@
+    }
+
+    /**
+     * Set Global Pointer and validate that its position does not fall in the GC trim range.
+     *
+     * If it falls we should throw a Trim Exception as this data no longer
+     * exists in the log and will be GC from all layers.
+     *
+     * @param globalPointer position to set the global pointer to.
+     */
+    public void setGlobalPointerCheckGCTrimMark(long globalPointer) {
+        validateGlobalPointerPosition(globalPointer);
+        this.setGlobalPointer(globalPointer);
+    }
+
+    /**
+     * Validate that Global Pointer position does not fall in the GC trim range.
+     *
+     * Note: we need to throw an exception whenever this is the case, as keeping active transactions
+     * in this range can lead to 'temporal' data loss when GC cycles are started, i.e., sync to an old version
+     * and trimming the resolved queue, before the updates between old version and trim mark are applied to the object.
+     *
+     * @param globalPointer position to set the global pointer to.
+     */
+    public void validateGlobalPointerPosition(long globalPointer) {
+        if (globalPointer < gcTrimMark && globalPointer > Address.NON_ADDRESS) {
+            throw new TrimmedException(String.format("Global pointer position[%s] is in GC trim range. " +
+                            "GC Trim mark: [%s]. This address is trimmed from the log.",
+                    globalPointer, gcTrimMark));
+        }
+    }
+
+    /** Add the given address to the resolved queue of the given context.
+     *
+     * @param globalAddress     The resolved global address.
+     */
+    public void addToResolvedQueue(long globalAddress) {
+        resolvedQueue.add(globalAddress);
+
+        if (maxResolution < globalAddress) {
+            maxResolution = globalAddress;
+        }
+    }
+
+    /**
+     * Reset all queues and pointers for this address space.
+     */
+    public void reset() {
+        globalPointer = Address.NON_ADDRESS;
+
+        readCpQueue.clear();
+        readQueue.clear();
+        resolvedQueue.clear();
+        minResolution = Address.NON_ADDRESS;
+        maxResolution = Address.NON_ADDRESS;
+
+        checkpointSuccessId = null;
+        checkpointSuccessStartAddr = Address.NEVER_READ;
+        checkpointSnapshotAddress = Address.NEVER_READ;
+    }
+
+    /** Seek to the requested global address. The next read will
+     * begin at the given global address, inclusive.
+     *
+     * @param globalAddress Address to seek to.
+     */
+    public synchronized void seek(long globalAddress) {
+        if (Address.nonAddress(globalAddress)) {
+            throw new IllegalArgumentException("globalAddress must"
+                    + " be >= Address.maxNonAddress()");
+        }
+        log.trace("Seek[{}]({}), min={} max={}", this,  globalAddress,
+                minResolution, maxResolution);
+        // Update minResolution if necessary
+        if (globalAddress >= maxResolution) {
```

Comment:
Because if the requested "seek address" is beyond our maximum resolved space (or the stream has not been accessed yet, so it is beyond -1), we need to move our min resolution to this "unknown space" otherwise, leaving the minResolution behind can lead to apparent data loss as these entries might not be synced yet (see the logic in previous or next in the AbstractQueuedStreamView).